File: backend/src/models/PublicCheckoutButton.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IPublicCheckoutButton extends Document {
  ownerId: mongoose.Types.ObjectId;
  buttonId: string;
  title: string;
  description?: string;
  defaultAmount?: number;
  allowCustomAmount: boolean;
  redirectUrl?: string;
  buttonText: string;
  buttonColor: string;
  isActive: boolean;
  totalClicks: number;
  totalPayments: number;
  createdAt: Date;
  updatedAt: Date;
}

const PublicCheckoutButtonSchema: Schema = new Schema({
  ownerId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
  buttonId: { type: String, required: true, unique: true },
  title: { type: String, required: true },
  description: { type: String },
  defaultAmount: { type: Number },
  allowCustomAmount: { type: Boolean, default: true },
  redirectUrl: { type: String },
  buttonText: { type: String, default: 'Pay with M-Pesa' },
  buttonColor: { type: String, default: '#25D366' },
  isActive: { type: Boolean, default: true },
  totalClicks: { type: Number, default: 0 },
  totalPayments: { type: Number, default: 0 },
}, {
  timestamps: true
});

PublicCheckoutButtonSchema.index({ buttonId: 1 });
PublicCheckoutButtonSchema.index({ ownerId: 1 });

export default mongoose.model<IPublicCheckoutButton>('PublicCheckoutButton', PublicCheckoutButtonSchema);

File: backend/app/models/public_checkout_button.py
from datetime import datetime, timezone
from typing import Optional
from beanie import Document, PydanticObjectId
from pydantic import Field


class PublicCheckoutButton(Document):
    owner_id: PydanticObjectId = Field(alias="ownerId")
    button_id: str = Field(unique=True, alias="buttonId")
    title: str
    description: Optional[str] = None
    default_amount: Optional[float] = Field(default=None, alias="defaultAmount")
    allow_custom_amount: bool = Field(default=True, alias="allowCustomAmount")
    redirect_url: Optional[str] = Field(default=None, alias="redirectUrl")
    button_text: str = Field(default="Pay with M-Pesa", alias="buttonText")
    button_color: str = Field(default="#25D366", alias="buttonColor")
    is_active: bool = Field(default=True, alias="isActive")
    total_clicks: int = Field(default=0, alias="totalClicks")
    total_payments: int = Field(default=0, alias="totalPayments")
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "publiccheckoutbuttons"
        indexes = [
            "button_id",
            "owner_id",
        ]
